sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -108,31 +118,41 @@ def test_kernel_pca_sparse():
             assert (X_pred_transformed.shape[1] ==
                          X_fit_transformed.shape[1])
 
-            # inverse transform
-            # X_pred2 = kpca.inverse_transform(X_pred_transformed)
-            # assert X_pred2.shape == X_pred.shape)
+            # inverse transform: not available for sparse matrices
+            with pytest.raises(NotFittedError):
+                kpca.inverse_transform(X_pred_transformed)
 
 
-def test_kernel_pca_linear_kernel():
+@pytest.mark.parametrize("solver", ["auto", "dense", "arpack", "randomized"],
+                         ids="solver={}".format)
+@pytest.mark.parametrize("n_features", [4, 10], ids="n_features={}".format)
```

Comment:
I think the default id is good enough here. (Same for the rest of the ids).

Reply:
As you prefer. I remove all of them.